### ISSUE TITLE
Nx "run-many build  --parallel" with Angular apps/libs projects takes a lot of CPU and Nx service is stopped...

### ISSUE THREAD

**Issue by tinesoft** [nrwl/nx#14343]
### Current Behavior

We have a Nx workspace with the following setup:

- 4 Angular applications
- 4 Angular libraries used in the above apps
- 1 storybook-host (library project, to show all storybooks)

![graph-anon](https://user-images.githubusercontent.com/4053092/212318565-538cf8ae-e547-4406-88bb-6e0ea9f37374.jpg)

When running the command to build all the apps/libs at once (minus the storybook),  the process takes a lot of CPU (almost 100%), and regularly  crashes on our CI/CD platform....

The crash almost always appears when building application "app4" ( and sometimes, but more rarely, "app3" or "app2").
The rest of applications and libraries gets build without errors most of the time.

Here is the relevant command :
```
> nx run-many --target build --parallel 4 --verbose --exclude storybook-host


 >  NX   Running target build for 8 project(s):

    - lib1
    - app4
    - app1
    - lib4
    - lib3
    - lib2
    - app3
    - app2

```

**Environment:** 

The build is launched on an **AWS Codebuild machine**, with the following steps:

- **Image**: aws/codebuild/standard:5.0
- **Environment type**: Linux
- **Compute Power**: 7 GB memory, 4 vCPUs (which is a lot imho...)

- **Angular** v14.2.10
- **Nx** v15.1.0

### Expected Behavior

The build should be successfull

### Steps to Reproduce

I cannot really provide repro steps...

### Nx Report

```shell
Node : 14.19.2
   OS   : linux x64
   npm  : 8.16.0
   
   nx : 15.1.0
   @nrwl/angular : 15.1.0
   @nrwl/cypress : 15.1.0
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.1.0
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.1.0
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.1.0
   @nrwl/js : 15.1.0
   @nrwl/linter : 15.1.0
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.1.0
   @nrwl/web : Not Found
   @nrwl/webpack : 15.1.0
   @nrwl/workspace : 15.1.0
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
     @fortawesome/angular-fontawesome: 0.11.1
     @nebular/theme: 9.0.3
     @ngrx/effects: 14.0.2
     @ngrx/router-store: 14.0.2
     @ngrx/store: 14.0.2
     @ngrx/schematics: 14.0.2
     @ngrx/store-devtools: 14.0.2
     @storybook/angular: 6.5.13
```

### Failure Logs
The main error is:

```shell
Error: Optimization error [main.756d983a00fcdf2b.js]: Error: The service was stopped
``` 

Here is the relevant part in the failure logs:

<details>
  <summary>Failure Log</summary>

```
> nx run app2:build:production

- Generating browser application bundles (phase: setup)...
    [webpack.cache.PackFileCacheStrategy] No pack exists at /codebuild/output/src956659751/src/workspace/.angular/cache/14.2.9/angular-webpack/d97e09ca500d5ca8e97cb5970b4b6eb66d6da5de.pack: Error: ENOENT: no such file or directory, stat '/codebuild/output/src956659751/src/workspace/.angular/cache/14.2.9/angular-webpack/d97e09ca500d5ca8e97cb5970b4b6eb66d6da5de/index.pack'
<t> [webpack.cache.PackFileCacheStrategy] restore cache container: 22.137155 ms
    [IdleFileCachePlugin] Initial cache was generated and cache will be persisted in 5s.
✔ Browser application bundle generation complete.
    [webpack.cache.PackFileCacheStrategy] Pack got invalid because of write to: ResolverCachePlugin|normal|default|dependencyType=|esm|path=|/codebuild/output/src956659751/src/workspace|request=|/codebuild/output/src956659751/src/workspace/apps/app2/src/main.ts
asset main.2c5153af9e6780c3.js 7.24 MiB {179} [cached] [immutable] (name: main)
asset styles.e7c4f4407aa432ef.css 350 KiB {532} [cached] [immutable] (name: styles)
asset 3rdpartylicenses.txt 48.5 KiB [cached]
asset polyfills.2b068d15a1e00cb3.js 34.4 KiB {429} [cached] [immutable] (name: polyfills)
asset 92.9a8d306a39a87099.js 11.1 KiB {92} [cached] [immutable] (name: myorg-lib3)
asset runtime.deecd9cc7a95072b.js 3.16 KiB {666} [cached] [immutable] (name: runtime)
asset 549.6dc2d43bf78baa0e.js 2.59 KiB {549} [cached] [immutable] (name: modules-dashboard-dashboard-module)
Entrypoint main = runtime.deecd9cc7a95072b.js main.2c5153af9e6780c3.js
Entrypoint polyfills = runtime.deecd9cc7a95072b.js polyfills.2b068d15a1e00cb3.js
Entrypoint styles = runtime.deecd9cc7a95072b.js styles.e7c4f4407aa432ef.css
chunk {92} (runtime: runtime) 92.9a8d306a39a87099.js (myorg-lib3) 58.8 KiB [rendered]
  ./libs/lib3/src/index.ts + 25 modules [3092] 58.8 KiB {92} [built] [code generated]
    import() @myorg/lib3 [496] ./apps/app2/src/main.ts + 9 modules ./apps/app2/src/app/app-routing.module.ts 13:22-52
chunk {179} (runtime: runtime) main.2c5153af9e6780c3.js (main) 7.02 MiB [initial] [rendered]
  dependent modules 6.8 MiB [dependent] 241 modules
  ./apps/app2/src/main.ts + 9 modules [496] 231 KiB {179} [built] [code generated]
    [no exports used]
     entry /codebuild/output/src956659751/src/workspace/apps/app2/src/main.ts main
chunk {429} (runtime: runtime) polyfills.2b068d15a1e00cb3.js (polyfills) 946 KiB [initial] [rendered]
  dependent modules 106 KiB [dependent] 1 module
  ./apps/app2/src/polyfills.ts + 3 modules [9282] 840 KiB {429} [built] [code generated]
    [no exports used]
    ModuleConcatenation bailout: Cannot concat with ./node_modules/zone.js/fesm2015/zone.js: Module is not an ECMAScript module
    entry /codebuild/output/src956659751/src/workspace/apps/app2/src/polyfills.ts polyfills
chunk {532} (runtime: runtime) styles.e7c4f4407aa432ef.css (styles) 50 bytes (javascript) 416 KiB (css/mini-extract) [initial] [rendered]
  dependent modules 416 KiB [dependent] 1 module
  ./apps/app2/src/styles.scss [7564] 50 bytes {532} [built] [code generated]
    [no exports used]
    entry /codebuild/output/src956659751/src/workspace/apps/app2/src/styles.scss styles
chunk {549} (runtime: runtime) 549.6dc2d43bf78baa0e.js (modules-dashboard-dashboard-module) 6.61 KiB [rendered]
  ./apps/app2/src/app/modules/dashboard/dashboard.module.ts + 3 modules [4549] 6.61 KiB {549} [built] [code generated]
    import() ./modules/dashboard/dashboard.module [496] ./apps/app2/src/main.ts + 9 modules ./apps/app2/src/app/app-routing.module.ts 9:22-68
chunk {666} (runtime: runtime) runtime.deecd9cc7a95072b.js (runtime) 8.54 KiB [entry] [rendered]
  runtime modules 8.54 KiB 15 modules


LOG from webpack.Compilation
<t> runtime requirements.modules: 0.724505 ms
<t> runtime requirements.chunks: 0.691838 ms
<t> runtime requirements.entries: 2.343137 ms
<t> compute affected modules: 0.319595 ms
<t> finish modules: 236.994976 ms
<t> report dependency errors and warnings: 61.29385 ms
<t> optimize dependencies: 255.035083 ms
<t> create chunks: 38.419954 ms
<t> compute affected modules with chunk graph: 0.179487 ms
<t> optimize: 567.559564 ms
    248 modules hashed, 0 from cache (0.42 variants per module in average)
<t> module hashing: 149.944441 ms
    100% code generated (248 generated, 0 from cache)
<t> code generation: 3056.923069 ms
<t> runtime requirements.modules: 39.717463 ms
<t> runtime requirements.chunks: 0.444246 ms
<t> runtime requirements.entries: 7.389666 ms
<t> runtime requirements: 47.779467 ms
<t> hashing: initialize hash: 0.015753 ms
<t> hashing: hash warnings: 0.005523 ms
<t> hashing: sort chunks: 0.15966 ms
<t> hashing: hash runtime modules: 9.276763 ms
<t> hashing: hash chunks: 5.286856 ms
<t> hashing: hash digest: 0.076002 ms
<t> hashing: process full hash modules: 0.002073 ms
<t> hashing: 15.732103 ms
    100% code generated (15 generated, 0 from cache)
<t> record hash: 0.100692 ms
<t> module assets: 0.368891 ms
<t> create chunk assets: 38.961726 ms
<t> process assets: 105787.103595 ms

LOG from webpack.Compiler
<t> make hook: 46274.919614 ms
<t> finish make hook: 0.278143 ms
<t> finish compilation: 299.122496 ms
<t> seal compilation: 109972.298933 ms
<t> afterCompile hook: 0.497798 ms

LOG from webpack.ResolverCachePlugin
    30% really resolved (1552 real resolves with 0 cached but invalid, 3544 cached valid, 0 concurrent)

LOG from webpack.FlagDependencyExportsPlugin
<t> restore cached provided exports: 4.778399 ms
<t> figure out provided exports: 161.039959 ms
    77% of exports of modules have been determined (136 no declared exports, 448 not cached, 1 flagged uncacheable, 0 from cache, 0 from mem cache, 13 additional calculations due to dependencies)
<t> store provided exports into cache: 19.596654 ms

LOG from webpack.InnerGraphPlugin
<t> infer dependency usage: 45.811249 ms

LOG from webpack.SideEffectsFlagPlugin
<t> update dependencies: 77.432048 ms

LOG from webpack.FlagDependencyUsagePlugin
<t> initialize exports usage: 24.364416 ms
<t> trace exports usage in graph: 151.886156 ms

LOG from webpack.buildChunkGraph
<t> visitModules: prepare: 23.239574 ms
<t> visitModules: visiting: 28.459766 ms
<t> visitModules: calculating available modules: 0.460787 ms
<t> visitModules: merging available modules: 0.452201 ms
<t> visitModules: check modules for revisit: 0.186249 ms
<t> visitModules: prepare: 1.514962 ms
<t> visitModules: visiting: 1.961099 ms
    932 queue items processed (431 blocks)
    2 chunk groups connected
    2 chunk groups processed for merging (2 module sets, 0 forked, 0 + 0 modules forked, 0 + 0 modules merged into fork, 0 resulting modules)
    2 chunk group info updated (0 already connected chunk groups reconnected)
<t> visitModules: 32.714563 ms
<t> connectChunkGroups: 0.292336 ms
<t> cleanup: 0.050564 ms

LOG from webpack.SplitChunksPlugin
<t> prepare: 0.055815 ms
<t> modules: 7.514488 ms
<t> queue: 0.009514 ms
<t> maxSize: 0.120365 ms

LOG from webpack.ModuleConcatenationPlugin
<t> select relevant modules: 4.779761 ms
<t> sort relevant modules: 0.417099 ms
<t> find modules to concatenate: 339.37045 ms
<t> sort concat configurations: 0.104313 ms
<t> create concatenated modules: 107.363065 ms
+ 3 hidden lines

LOG from build-angular.JavaScriptOptimizerPlugin
<t> optimize asset: 549.161f32f4255b3256.js: 2168.92261 ms
<t> optimize asset: runtime.df1078d05c2b5b48.js: 2264.6824500000002 ms
<t> optimize asset: 92.d903190f9b403aeb.js: 2797.771543 ms
<t> optimize asset: polyfills.935057314d95edab.js: 72980.147157 ms
<t> optimize js assets: 73556.90745 ms

LOG from build-angular.CssOptimizerPlugin
<t> optimize asset: styles.cbd3643584215411.css: 111.072957 ms
<t> optimize css assets: 118.035649 ms

LOG from webpack.FileSystemInfo
    2132 new snapshots created
    0% root snapshot uncached (0 / 3544)
    0% children snapshot uncached (0 / 0)
    0 entries tested
    File info in cache: 1423 timestamps 337 hashes 337 timestamp hash combinations
    File timestamp snapshot optimization: 40% (338/849) entries shared via 18 shared snapshots (39 times referenced)
    File timestamp hash combination snapshot optimization: 83% (12792/15369) entries shared via 67 shared snapshots (1754 times referenced)
    Directory info in cache: 0 timestamps 0 hashes 0 timestamp hash combinations
    Missing items snapshot optimization: 72% (5715/7992) entries shared via 147 shared snapshots (1191 times referenced)
    Managed items info in cache: 189 items
    Managed items snapshot optimization: 43% (1860/4339) entries shared via 35 shared snapshots (354 times referenced)
    Managed files snapshot optimization: 73% (5992/8253) entries shared via 213 shared snapshots (1389 times referenced)
    Managed contexts snapshot optimization: 0% (0/1) entries shared via 0 shared snapshots (0 times referenced)
    Managed missing snapshot optimization: 77% (6267/8189) entries shared via 262 shared snapshots (1131 times referenced)

WARNING in /codebuild/output/src956659751/src/workspace/node_modules/@angular/material-moment-adapter/fesm2020/material-moment-adapter.mjs depends on 'moment'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies
Error: /codebuild/output/src956659751/src/workspace/node_modules/@angular/material-moment-adapter/fesm2020/material-moment-adapter.mjs depends on 'moment'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies
    at addWarning (/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/utils/webpack-diagnostics.js:13:31)
    at /codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/webpack/plugins/common-js-usage-warn-plugin.js:75:70
    at _next0 (eval at create (/codebuild/output/src956659751/src/workspace/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:57:1)
    at eval (eval at create (/codebuild/output/src956659751/src/workspace/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:100:1)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)

ERROR in Optimization error [main.756d983a00fcdf2b.js]: Error: The service was stopped
    at /codebuild/output/src956659751/src/workspace/node_modules/esbuild/lib/main.js:1400:29
    at /codebuild/output/src956659751/src/workspace/node_modules/esbuild/lib/main.js:678:9
    at Socket.afterClose (/codebuild/output/src956659751/src/workspace/node_modules/esbuild/lib/main.js:656:7)
    at Socket.emit (events.js:412:35)
    at endReadableNT (internal/streams/readable.js:1333:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
Error: Optimization error [main.756d983a00fcdf2b.js]: Error: The service was stopped
    at /codebuild/output/src956659751/src/workspace/node_modules/esbuild/lib/main.js:1400:29
    at /codebuild/output/src956659751/src/workspace/node_modules/esbuild/lib/main.js:678:9
    at Socket.afterClose (/codebuild/output/src956659751/src/workspace/node_modules/esbuild/lib/main.js:656:7)
    at Socket.emit (events.js:412:35)
    at endReadableNT (internal/streams/readable.js:1333:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
    at /codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/webpack/plugins/javascript-optimizer-plugin.js:154:55
    at async Promise.all (index 4)
    at async /codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/webpack/plugins/javascript-optimizer-plugin.js:158:21

2023-01-13 11:13:58: webpack 5.74.0 compiled in 171863 ms (f559f92bd94b0ba7)
    [webpack.cache.PackFileCacheStrategy] Storing pack...
    [webpack.cache.PackFileCacheStrategy] Capturing build dependencies... ()
<t> [webpack.cache.PackFileCacheStrategy] resolve build dependencies: 5.653684 ms
<t> [webpack.cache.PackFileCacheStrategy] snapshot build dependencies: 3.900818 ms
    [webpack.cache.PackFileCacheStrategy] Captured build dependencies
    [webpack.cache.PackFileCacheStrategy] 3280 fresh items in cache put into pack 0 (2552 items), 1 (720 items), 2 (8 items)
<i> [webpack.cache.PackFileCacheStrategy] Serialization of 'ResolverCachePlugin|normal|default|dependencyType=|esm|path=|/codebuild/output/src956659751/src/workspace|request=|/codebuild/output/src956659751/src/workspace/libs/lib1/src/lib/table/pagination/pagination.component.scss?ngResource': 10.960702 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'ResolverCachePlugin|loader|default|path=|/codebuild/output/src956659751/src/workspace|request=|/codebuild/output/src956659751/src/workspace/node_modules/sass-loader/dist/cjs.js': 5.546562 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/mini-css-extract-plugin/dist/loader.js!/codebuild/output/src956659751/src/workspace/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[5].rules[0].oneOf[0].use[1]!/codebuild/output/src956659751/src/workspace/node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[5].rules[0].oneOf[0].use[2]!/codebuild/output/src956659751/src/workspace/node_modules/resolve-url-loader/index.js??ruleSet[1].rules[5].rules[1].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[5].rules[1].use[1]!/codebuild/output/src956659751/src/workspace/apps/app2/src/styles.scss': 2.225108 ms
<i> [webpack.cache.PackFileCacheStrategy] Serialization of 'angular-compiler:resource|32|Compilation/modules|asset/source|/codebuild/output/src956659751/src/workspace/node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[5].rules[0].oneOf[1].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/resolve-url-loader/index.js??ruleSet[1].rules[5].rules[1].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[5].rules[1].use[1]!/codebuild/output/src956659751/src/workspace/libs/lib2/src/lib/sidebar/sidebar-export/sidebar-export.component.scss?ngResource': 13.876325 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/apps/app2/src/main.ts': 3.840925 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/apps/app2/src/polyfills.ts': 5.738355 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/platform-browser/fesm2020/platform-browser.mjs': 2.191599 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/core/fesm2020/core.mjs': 2.981387 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'ResolverCachePlugin|normal|default|fullySpecified=|false|dependencyType=|esm|path=|/codebuild/output/src956659751/src/workspace/node_modules/@angular/platform-browser/fesm2020|request=|@angular/common': 1.427705 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'ResolverCachePlugin|normal|default|fullySpecified=|false|dependencyType=|esm|path=|/codebuild/output/src956659751/src/workspace/node_modules/@angular/platform-browser/fesm2020|request=|@angular/core': 5.841468 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/rxjs/dist/esm/operators/index.js': 2.605193 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/rxjs/dist/esm/index.js': 4.965298 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/libs/lib2/src/index.ts': 1.946787 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/router/fesm2020/router.mjs': 4.579091 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/common/fesm2020/common.mjs': 3.777696 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/apps/app2/src/app/shared/guards/is-logged-local.guard.ts': 2.386704 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/libs/lib2/src/lib/sidebar/sidebar.module.ts': 1.927269 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/libs/lib2/src/lib/auth/forgot-password/forgot-password.component.ts': 1.011951 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/libs/lib2/src/lib/auth/login/login.component.ts': 4.03157 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/libs/lib2/src/lib/auth/renew-password/renew-password.component.ts': 1.49633 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/libs/lib2/src/lib/sidebar/sidebar-export/sidebar-export.component.ts': 3.796844 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/libs/lib2/src/lib/sidebar/sidebar-outline/sidebar-outline.component.ts': 1.073469 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/libs/lib2/src/lib/user/user-profile/user-profile.component.ts': 1.919917 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/libs/lib2/src/lib/api/local-token-service.ts': 2.607596 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/rxjs/dist/esm/internal/operators/pluck.js': 4.070311 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@nebular/auth/fesm2020/nebular-auth.mjs': 6.297542 ms
<i> [webpack.cache.PackFileCacheStrategy] Serialization of 'ResolverCachePlugin|normal|default|fullySpecified=|false|dependencyType=|esm|path=|/codebuild/output/src956659751/src/workspace/node_modules/rxjs/dist/esm|request=|./internal/operators/auditTime': 17.886497 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/material/fesm2020/core.mjs': 1.162324 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/cdk/fesm2020/overlay.mjs': 1.100662 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/cdk/fesm2020/a11y.mjs': 1.077482 ms
<i> [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@fortawesome/free-solid-svg-icons/index.es.js': 13.081671 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/forms/fesm2020/forms.mjs': 2.10248 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/compiler/fesm2020/compiler.mjs': 1.250147 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'ResolverCachePlugin|normal|default|fullySpecified=|false|dependencyType=|esm|path=|/codebuild/output/src956659751/src/workspace/node_modules/rxjs/dist/esm/internal/operators|request=|../Subscriber': 8.88395 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/cdk/fesm2020/scrolling.mjs': 1.41245 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'ResolverCachePlugin|normal|default|fullySpecified=|false|dependencyType=|esm|path=|/codebuild/output/src956659751/src/workspace/libs/lib1/src/lib/basic-dropdown-list|request=|@angular/core': 3.633482 ms
<i> [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@nebular/theme/fesm2020/nebular-theme.mjs': 26.187335 ms
<i> [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/material/fesm2020/autocomplete.mjs': 16.708358 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/material/fesm2020/datepicker.mjs': 5.012224 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/material/fesm2020/form-field.mjs': 1.743625 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/cdk/fesm2020/table.mjs': 1.765653 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/modules|/codebuild/output/src956659751/src/workspace/node_modules/moment/locale|sync|/^\.\/.*$/': 1.836243 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|/codebuild/output/src956659751/src/workspace/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[5].rules[0].oneOf[0].use[1]!/codebuild/output/src956659751/src/workspace/node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[5].rules[0].oneOf[0].use[2]!/codebuild/output/src956659751/src/workspace/node_modules/resolve-url-loader/index.js??ruleSet[1].rules[5].rules[1].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[5].rules[1].use[1]!/codebuild/output/src956659751/src/workspace/apps/app2/src/styles.scss|build time': 2.760654 ms
<i> [webpack.cache.PackFileCacheStrategy] Serialization of 'FlagDependencyExportsPlugin|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/core/fesm2020/core.mjs': 14.490146 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'FlagDependencyExportsPlugin|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/cdk/fesm2020/a11y.mjs': 1.475862 ms
<i> [webpack.cache.PackFileCacheStrategy] Serialization of 'FlagDependencyExportsPlugin|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@fortawesome/free-solid-svg-icons/index.es.js': 24.066062 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'FlagDependencyExportsPlugin|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@fortawesome/free-regular-svg-icons/index.es.js': 1.364709 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'FlagDependencyExportsPlugin|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/compiler/fesm2020/compiler.mjs': 1.914334 ms
<i> [webpack.cache.PackFileCacheStrategy] Serialization of 'FlagDependencyExportsPlugin|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/rxjs/dist/esm/internal/util/reportUnhandledError.js': 12.035891 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'FlagDependencyExportsPlugin|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@nebular/theme/fesm2020/nebular-theme.mjs': 2.849027 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/core/fesm2020/core.mjs|runtime': 8.07994 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/common/fesm2020/common.mjs|runtime': 2.480939 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/cdk/fesm2020/overlay.mjs|runtime': 1.169303 ms
<i> [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@fortawesome/free-solid-svg-icons/index.es.js|runtime': 16.835208 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/forms/fesm2020/forms.mjs|runtime': 2.289325 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@fortawesome/free-regular-svg-icons/index.es.js|runtime': 1.202635 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/libs/lib1/src/index.ts|1002991bd099d6e3|runtime': 7.030785 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/libs/lib2/src/index.ts|b580e38fd0fbc433|runtime': 9.603078 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/libs/lib3/src/index.ts|932754b506b0c85d|runtime': 1.851077 ms
<i> [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@nebular/auth/fesm2020/nebular-auth.mjs|9f023584cc1f38ce|runtime': 18.803487 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/cdk/fesm2020/scrolling.mjs|443ba4c1f1b350f0|runtime': 5.009645 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/apps/app2/src/main.ts|3b4bc9bfe1cb3caa|runtime': 2.53437 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@angular/router/fesm2020/router.mjs|c9884c14380faa67|runtime': 2.676082 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@ngtools/webpack/src/ivy/index.js!/codebuild/output/src956659751/src/workspace/apps/app2/src/polyfills.ts|1ea381c174a284f4|runtime': 8.749104 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/codeGeneration|javascript/esm|/codebuild/output/src956659751/src/workspace/node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js??ruleSet[1].rules[2].use[0]!/codebuild/output/src956659751/src/workspace/node_modules/@fortawesome/angular-fontawesome/fesm2020/angular-fontawesome.mjs|034e76a6ad4bb20a|runtime': 1.373142 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/assets|chunk179': 8.183767 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/assets|chunk429': 4.810357 ms
    [webpack.cache.PackFileCacheStrategy] Serialization of 'Compilation/assets|mini-css-extract-plugin.532': 2.458425 ms
<t> [webpack.cache.PackFileCacheStrategy] store pack: 1489.430123 ms
    [webpack.cache.PackFileCacheStrategy] Stored pack (3280 items, 3 files, 113 MiB)
✔ Browser application bundle generation complete.

Warning: /codebuild/output/src956659751/src/workspace/node_modules/@angular/material-moment-adapter/fesm2020/material-moment-adapter.mjs depends on 'moment'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies



Error: Optimization error [main.756d983a00fcdf2b.js]: Error: The service was stopped

```
</details>
```


### Additional Information


**Note1:**

Here is the result of the **profiling of a build that was successful** (on same AWS machine)...
```
NX_PROFILE=profile.json npx nx run-many --target build --parallel 4 --exclude storybook-host
``` 

<details>
  <summary>profile.json</summary>

```
[
  {
    "name": "thread_name",
    "ph": "M",
    "pid": 503,
    "tid": 0,
    "ts": 0,
    "args": {
      "name": "Group #1"
    }
  },
  {
    "name": "lib1:build:production",
    "cat": "lib1,build,production",
    "ph": "X",
    "ts": 663338.1169999484,
    "dur": 6234071.524000028,
    "pid": 503,
    "tid": 0,
    "args": {
      "target": {
        "project": "lib1",
        "target": "build",
        "configuration": "production"
      },
      "status": "success"
    }
  },
  {
    "name": "thread_name",
    "ph": "M",
    "pid": 503,
    "tid": 1,
    "ts": 0,
    "args": {
      "name": "Group #2"
    }
  },
  {
    "name": "thread_name",
    "ph": "M",
    "pid": 503,
    "tid": 2,
    "ts": 0,
    "args": {
      "name": "Group #3"
    }
  },
  {
    "name": "lib4:build:production",
    "cat": "lib4,build,production",
    "ph": "X",
    "ts": 6897981.685999956,
    "dur": 5626896.502999997,
    "pid": 503,
    "tid": 2,
    "args": {
      "target": {
        "project": "lib4",
        "target": "build",
        "configuration": "production"
      },
      "status": "success"
    }
  },
  {
    "name": "lib2:build:production",
    "cat": "lib2,build,production",
    "ph": "X",
    "ts": 6897943.1110000005,
    "dur": 7847804.744999972,
    "pid": 503,
    "tid": 1,
    "args": {
      "target": {
        "project": "lib2",
        "target": "build",
        "configuration": "production"
      },
      "status": "success"
    }
  },
  {
    "name": "thread_name",
    "ph": "M",
    "pid": 503,
    "tid": 3,
    "ts": 0,
    "args": {
      "name": "Group #4"
    }
  },
  {
    "name": "lib3:build:production",
    "cat": "lib3,build,production",
    "ph": "X",
    "ts": 14746058.65099997,
    "dur": 8067561.134000018,
    "pid": 503,
    "tid": 0,
    "args": {
      "target": {
        "project": "lib3",
        "target": "build",
        "configuration": "production"
      },
      "status": "success"
    }
  },
  {
    "name": "app4:build:production",
    "cat": "app4,build,production",
    "ph": "X",
    "ts": 14746102.715999994,
    "dur": 159524524.19700003,
    "pid": 503,
    "tid": 2,
    "args": {
      "target": {
        "project": "app4",
        "target": "build",
        "configuration": "production"
      },
      "status": "success"
    }
  },
  {
    "name": "app2:build:production",
    "cat": "app2,build,production",
    "ph": "X",
    "ts": 22814099.29699998,
    "dur": 158399007.40699995,
    "pid": 503,
    "tid": 0,
    "args": {
      "target": {
        "project": "app2",
        "target": "build",
        "configuration": "production"
      },
      "status": "success"
    }
  },
  {
    "name": "app3:build:production",
    "cat": "app3,build,production",
    "ph": "X",
    "ts": 14746148.042999964,
    "dur": 173862515.236,
    "pid": 503,
    "tid": 3,
    "args": {
      "target": {
        "project": "app3",
        "target": "build",
        "configuration": "production"
      },
      "status": "success"
    }
  },
  {
    "name": "app1:build:production",
    "cat": "app1,build,production",
    "ph": "X",
    "ts": 22813922.112,
    "dur": 170138753.28,
    "pid": 503,
    "tid": 1,
    "args": {
      "target": {
        "project": "app1",
        "target": "build",
        "configuration": "production"
      },
      "status": "success"
    }
  }
]
```

</details>

**Note 2:**

I have tried to increase the memory allocated to the Node process,  by adding a **.build.env** env file with below content, but it didn't help much...

```
NODE_OPTIONS=--max-old-space-size=4096
```
> content of the **.build.env** file located at workspace root

**Note 3:**

Running, the exact same build command on my local computer, also shows high CPU usage, but never fails...
- CPU: Intel(R) Xeon(R) CPU E3-1505M v6 @ 3.00GHz   3.00 GHz
- RAM: 16.0 GB (15.9 GB usable)
- OS: Windows 10 Enterprise, 64 Bits

**Comment by leosvelperez** [comment]
hey @tinesoft!

Have you been able to address the issue?

This is a tricky one, given it might be env specific and we don't have a repro for it. Please note the issue comes from the `esbuild`-based optimizer the Angular CLI builders use. From what I can see, if I had to guess something, it could be a thread exhaustion issue. You are running the command with 4 parallel processes and at the same time, the `esbuild`-based optimizer plugin creates (or tries to create) up to 4 threads (by default) per app build to optimize an app's scripts.

You could try reducing the number of parallel tasks you're running or providing the env variable `NG_BUILD_MAX_WORKERS` with a number lower than 4. It might be a case where you need to find the right combination for the available CPUs in your CI.

If this doesn't work, we would need a repro project that triggers the issue when used in a similar env that you described. Could you please update us with your findings?

**Comment by ihor-panasiuk95** [comment]
I have the same issue but in Github Actions.
I have about 13 apps and they are build within 6 matrix jobs (like 2-3 per job, they are build in parallel)
Build randomly fails, usually on the same application. Re-running build usually helps.
That's what I see in NX Cloud for failed target:
![image](https://user-images.githubusercontent.com/69238797/218983978-3747e0fe-391a-4803-8c96-9371e72c21dc.png)

And needs to admit, it suddenly happed few weeks ago. Before, we had no issues with it. I guess some updated dependency (in package-lock) introduced this but can't identify what exactly.

**Comment by ihor-panasiuk95** [comment]
Adding more specifics:
Angular 15 (15.0.1)
NX 15 (15.2.3)
Github Actions

**Comment by tinesoft** [comment]
Hi @leosvelperez 

Thank you for your answer (and sorry for the late reply).

I did follow your advice (trying to avoid the thread exhaustion) , but yet, I was still able to reproduce the issue.
Although, it seems to appear much less than before...
.

As suggested, I reduced the number of workers allocated to the NG_BUILD, based on the number of available CPUs on the host (AWS) machine running the `nx build` command

The actual command that was run is: `env-cmd -f .build.env.mjs nx run-many --target build --parallel 4`

where **.build.env.mjs** is a simple  script to compute the number of workers to dedicate to NG_BUILD ( = Nb CPUs -1 ), that is then exposed via an environment variable via the `env-cmd` helper tool, as suggested in [Nx docs](https://nx.dev/recipes/environment-variables/define-environment-variables#pointing-to-custom-env-files)

**.build.env.mjs**
```
import * as os from 'os';
import * as fs from 'fs';


// Adapted from https://github.com/xiaoxiaojx/get_cpus_length
 function getCpuLimits() {
	const readNumber = (path) => Number(fs.readFileSync(file, "utf-8"));
  try {
		// Only for instances using CFS scheduling strategy https://www.kernel.org/doc/Documentation/scheduler/sched-bwc.txt
    // The cfs.cpu_period_us file records the scheduling period, the unit is us; the default value is generally 100’000, that is, 100 ms
    // cfs.cpu_quota_us records the amount of cpu allowed to be used by a process in each scheduling cycle, and the unit is also us. A value of -1 means unlimited; for 4C containers, this value is typically 400’000
    return (
      readNumber("/sys/fs/cgroup/cpu/cpu.cfs_quota_us") /
      readNumber("/sys/fs/cgroup/cpu/cpu.cfs_period_us")
    );
  } catch (err) {
    return -1;
  }
}

// https://github.com/nodejs/node/issues/28762
const nbAvailableCpus = getCpuLimits() || os.cpus.length;

// https://github.com/nrwl/nx/issues/14343#issuecomment-1428212730
const ngBuildMaxWorkers = Math.min(1, nbAvailableCpus - 1);

console.log(`NOTE: The NG_BUILD will use up to ${ngBuildMaxWorkers} workers max, based on the number of CPUs available(${nbAvailableCpus})`);

export default {
// Increase memory allocated to the node process during the build, to avoid OOM
// This will also reduce the execution time for the build almost by half
	NODE_OPTIONS : "--max-old-space-size=4096",
	NG_BUILD_MAX_WORKERS: ngBuildMaxWorkers
}

```



I wonder if I should also remove the `--parallel 4` part in the above command and make it dynamic? (since the issue is on NG_BUILD, not Nx side apparently) 

Thanks in advance
